tangram/src/lib.rs

```diff
@@ -0,0 +1,559 @@
+	/// pub ItemMetadata get(fn item_metadata): map hasher(identity) T::Hash => TangramMetadata;
+	#[pallet::storage]
+	#[pallet::getter(fn item_metadata)]
+	pub(super) type ItemMetadata<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, TangramItemMetadata, ValueQuery>; // Todo identity
+
+	/// All Items associated with an account
+	/// pub ItemsForAccount get(fn items_for_account): map hasher(blake2_128_concat) T::AccountId => Vec<T::Hash>;
+	#[pallet::storage]
+	#[pallet::getter(fn items_for_account)]
+	pub(super) type ItemsForAccount<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, Vec<T::Hash>, ValueQuery>; 
+
+	/// Owner of an Item
+	///	pub AccountForItem get(fn account_for_item): map hasher(blake2_128_concat) T::Hash => T::AccountId;
+	#[pallet::storage]
+	#[pallet::getter(fn account_for_item)]
+	pub(super) type AccountForItem<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::AccountId, ValueQuery>; 
+
+
+	/// Retrieve an Item Hash by its indexes
+	/// pub ItemByIndex get(fn item_by_index): map hasher(blake2_128_concat) (RealmIndex,ClassIndex,ItemIndex) => T::Hash;
+	#[pallet::storage]
+	#[pallet::getter(fn item_by_index)]
+	pub(super) type ItemByIndex<T: Config> = StorageMap<_, Blake2_128Concat, (RealmIndex,ClassIndex,ItemIndex), T::Hash, ValueQuery>; 
+
+
+
+	// All items?
+	//pub TotalForAccount get(fn total_for_account): map hasher(blake2_128_concat) T::AccountId => u64;
+	#[pallet::storage]
+	#[pallet::getter(fn total_for_account)]
+	pub(super) type TotalForAccount<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, u64, ValueQuery>; 
+
+
+
+	/// Total Token in system
+	/// pub Total get(fn total_items): TotalIndex;
+	#[pallet::storage]
+	#[pallet::getter(fn total_items)]
+	pub(super) type Total<T: Config> = StorageValue<_, TotalIndex, ValueQuery>; 
+
+
+
+	/// Burned Token in system
+	/// pub Burned get(fn burend_items): BurnedIndex;
+	#[pallet::storage]
+	#[pallet::getter(fn burned_items)]
+	pub(super) type Burned<T: Config> = StorageValue<_, BurnedIndex, ValueQuery>; 
+
+
+
+//-----------------------------news 
+
+
+	impl<Hash> TangramRealm <Hash> {
+		pub fn new(id: Hash, org: Hash, index: u64) ->  TangramRealm<Hash> where Hash: Clone, {
+			TangramRealm {
+				id,
+				org: org,
+				index: index,
+			}
+		}
+	}
+
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+	
+
+		#[pallet::weight(<T as Config>::WeightInfo::create_realm())]
+		pub fn create_realm(
+			origin: OriginFor<T>,
+			org: T::Hash
+		) -> DispatchResult {
+			// TODO: ensure origin == controller
+			// ensure_root(origin)?;
+
+			// TODO: ensure org exists : How???
+			// TODO: ensure org does not have a realm yet : ? Can org have 2 realms???
+			// ensure!(!<TangramRealm<T>>::contains_key(&account), Error::<T>::EntityExists);
+
+			let index = <NextRealmIndex<T>>::get();
+			let (hash, _blockNumber) = <T as Config>::Randomness::random(b"rndrealm");
+			let realm = TangramRealm::new(hash.clone(), org, index.clone() );
```

Comment:
Is there a chance to avoid `new` method? This struct is located in the same privacy scope as current pallet so I think there should be no issues of calling constructor directly.